renter/payment.go

```diff
@@ -0,0 +1,146 @@
+package renter
+
+import (
+	"crypto/ed25519"
+	"errors"
+	"fmt"
+
+	"go.sia.tech/core/net/mux"
+	"go.sia.tech/core/net/rhp"
+	"go.sia.tech/core/net/rpc"
+	"go.sia.tech/core/types"
+	"lukechampine.com/frand"
+)
+
+type (
+	// A PaymentMethod pays for RPC usage during an renter-host session.
+	PaymentMethod interface {
+		Pay(s *mux.Stream, amount types.Currency) error
```

Comment:
this is the sort of thing I was talking about earlier -- feels wrong to abstract stream side-effects via an interface. This should be `Pay(amount types.Currency) (something, error)`

Reply:
I think the only other way this would work would be changing the interface to `IsPaymentMethod()`, then switching on the type in a concrete `processPayment`, but that seems worse. The contract payment method requires a back and forth, so I can't just return an `rpc.Object` and write it to the stream.